### ISSUE TITLE
Tox-Sidebar Role=Complementary Accessibility Bug

### ISSUE THREAD

**Issue by toni-carlson** [tinymce/tinymce#7639]
**What is the current behavior? Describe the bug**
There is an accessibility bug with the `.tox-sidebar` element having the role of `complementary`.

I researched more about TinyMCE's use of this sidebar and its function. It appears that its only function is for when the user has enabled the Comments plugin which is only an option for paid TinyMCE subscribers. Once the user enables the Comments plugin, there’s a menu option to add and show/hide the comments which opens/closes this `tox-sidebar`, but the `.tox-sidebar` is in the DOM even when users have not enabled this plugin.

In addition, for when the element is correctly in the DOM, the `complementary` role's purpose is to give supporting information for the main content of a webpage as a whole and is not directly related to interacting with the main content of the page (meaning it could technically stand on its own). This sidebar is directly related to the TinyMCE editor, which means it's breaking two aspects of this role: not related to the main content of the whole webpage, and directly related to the content it's beside.

**Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud/) or similar.**
When running axe DevTools, we received the following error:
![image](https://user-images.githubusercontent.com/66501530/154584604-508f30d0-29c0-4cac-8e0a-2dbe9bd62a32.png)

**What is the expected behavior?**
Since not all users are paid TinyMCE subscribers, and those that are still may not enable the Comments plugin, it's best to avoid having the `.tox-sidebar` in the DOM unless the Comments plugin is enabled. This can also slightly enhance the performance of TinyMCE.

I think a role of `region` might be the best fit for this sidebar instead. I have verified changing the role to `region` will pass the auditor check with axe DevTools.

**TLDR:** The `.tox-sidebar` is only in the DOM if the Comments plugin is enabled, and the `.tox-sidebar` role is `region`.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
Current version, all browsers & OS, not sure if previous versions had this issue or not

**Comment by MitchCTiny** [comment]
Hi @toni-carlson 
The sidebar element can be configured via the use of the Sidebar API, documented here:
https://www.tiny.cloud/docs/tinymce/6/customsidebar/
Any integrator can make use of the sidebar without requiring the Comments plugin.

The role attribute of the sidebar is incorrect, and has been logged as a bug.
Thank you for finding this.

**Comment by shanmen-tiny** [comment]
The fix will be available in the next release. Thank you.
